Arrowgene.Ddon.GameServer/Handler/DispelExchangeDispelItemHandler.cs

```diff
@@ -0,0 +1,154 @@
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.GameServer.Dump;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Model.Appraisal;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class DispelExchangeDispelItemHandler : GameRequestPacketHandler<C2SDispelExchangeDispelItemReq, S2CDispelExchangeDispelItemRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(DispelExchangeDispelItemHandler));
+
+        public DispelExchangeDispelItemHandler(DdonGameServer server) : base(server)
+        {
+        }
+
+        public override S2CDispelExchangeDispelItemRes Handle(GameClient client, C2SDispelExchangeDispelItemReq request)
+        {
+            var res = new S2CDispelExchangeDispelItemRes();
+
+            var appraisialItems = Server.AssetRepository.SpecialShopAsset.AppraisalItems;
+            foreach (var item in request.GetDispelItemList)
+            {
+                if (!appraisialItems.ContainsKey(item.Id))
+                {
+                    // TODO: Throw an exception?
+                    continue;
+                }
+
+                bool toBag = false;
+                switch (item.StorageType)
+                {
+                    case 19:
+                        toBag = true;
+                        break;
+                    case 20:
+                        toBag = false;
+                        break;
+                    default:
+                        throw new Exception($"Unexpected destination when exchanging items {item.StorageType}");
```

Comment:
It's better to use a ResponseErrorException so the user gets a more detailed error code